src/conn/conn_compact.c

```diff
@@ -602,6 +620,11 @@ __background_compact_server(void *arg)
         if (!running)
             continue;
 
+        /* Check if there is cache pressure. */
+        cache_pressure = __wt_cache_bytes_inuse(cache) >= cache_pressure_limit;
```

Comment:
Should we consider dirty eviction threshold and update threshold as well?

Reply:
We tried this and unfortunately, it's very easy to hit those thresholds while the system is not in trouble and eviction is doing its job. We thought that checking if we have a cache almost full would be the best way to detect that "the system might be in trouble soon"

Reply:
You mean it is easy to reach dirty and update trigger?

Reply:
Yes